cpp/arcticdb/storage/s3/s3_storage.hpp

```diff
@@ -245,7 +245,9 @@ auto get_s3_config(const ConfigType& conf) {
     auto endpoint = conf.endpoint();
     util::check_arg(!endpoint.empty(), "S3 Endpoint must be specified");
     client_configuration.endpointOverride = endpoint;
-    client_configuration.verifySSL = false;
+    const bool use_ssl = ConfigsMap::instance()->get_int("S3Storage.UseSSL", conf.ssl());
```

Comment:
Will this cast the int to a bool?! What if I set a value that's neither 0 or 1?

Reply:
I think it's casting, if anything another number is passed it will be set to true. I didn't find a flag option. There were get_int, get_double, get_string. Should I make get bool?

Reply:
It is indeed casting. But the whole thin is implemented via macro: https://github.com/man-group/ArcticDB/blob/da1d286e8a6840a93218ba3bede4e9fc661cb59d/cpp/arcticdb/util/configs_map.hpp#L34 even if I simply add bool it will cast it the same way. So the questions are: 

1. What do we want the functionality to be if neither 0 nor 1 is passed?
2. Do we want to add something as "check_flag" which only checks if the map contains the key without reading the value?

Also I really don't like the macro design of this class. IMO it can be replaced by template member functions.